src/device-manager/java/src/nl/Weave/DeviceManager/WirelessRegulatoryConfig.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *
+ *    Copyright (c) 2019 Google LLC
+ *    Copyright (c) 2013-2017 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package nl.Weave.DeviceManager;
+
+/**
+ * Container for wireless regulatory configuration information.
+ */
+public class WirelessRegulatoryConfig
+{
+    /** Active wireless regulatory domain.
+     */
+    public String RegDomain;
```

Comment:
Can we refactor this to make it immutable? Something like this:

```
public final class WirelessRegulatoryConfig {
  private final String regulatoryDomain;
  private final WirelessOperatingLocation operatingLocation;
  private final List<String> supportedRegulatoryDomains;

  public WirelessRegulatoryConfig(
      String regulatoryDomain, 
      WirelessOperatingLocation operatingLocation, 
      List<String> supportedRegulatoryDomains) {
        this.regulatoryDomain = regulatoryDomain;
        this.operatingLocation = operatingLocation;
        this.supportedRegulatoryDomains = new ArrayList<>(supportedRegulatoryDomains);
  }

  public WirelessOperatingLocation getOperatingLocation() { 
    return operatingLocation;
  }

  public String getRegulatoryDomain() {
    return regulatoryDomain;
  }

  public List<String> getSupportedRegulatoryDomains() {
    return new ArrayList<>(supportedRegulatoryDomains);
  }
}
```

This also avoids the need for a factory method (`Make`) although you could have both. This guarantees that all fields are initialized, though, and cannot be modified.

Reply:
Making the fields private requires considerable re-working of the JNI code.  I could add the getters and the constructor, and then make the fields package scope final.  I would still need Make(), but that could be package scope as well.  Thoughts?

Reply:
Chatted offline, didn't realize this would require a larger refactor of the JNI code. Avoiding the getters and just having public final fields should be fine.

Reply:
My latest commit makes the class immutable but preserves the existing public fields.